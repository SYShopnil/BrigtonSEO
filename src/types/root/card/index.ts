export interface ICard {
  dataTestId: string;
  cardShadow: CardShadowEnum;
  hoverCardShadow?: CardShadowEnum;
  cardBorderRadius: CardBorderRadiusEnum;
}

export enum CardShadowEnum {
  ShadowOne = 'shadow-One',
  ShadowTwo = 'shadow-Two',
  ShadowThree = 'shadow-Three',
  None = 'shadow-none',
}

export enum CardBorderRadiusEnum {
  XL = 'radius-xl',
  '2XL' = 'radius-2xl',
}
