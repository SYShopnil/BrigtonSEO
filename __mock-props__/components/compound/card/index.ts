import { eventHistoryCardData } from './event-history';
import { testimonialData, testimonialWithoutImageData } from './testimonial';
import {
  speakerCardNormalData,
  speakerCardTransparentData,
  teamData,
} from './person';
import { courseCardData } from './course';
import { upcomingCardData } from './upcoming';
import { pricingCardLargeData, pricingCardSmallData } from './pricing-card';
import { buyTicketCardData } from './buy-ticket';

export {
  testimonialData,
  pricingCardLargeData,
  pricingCardSmallData,
  courseCardData,
  upcomingCardData,
  speakerCardNormalData,
  speakerCardTransparentData,
  teamData,
  eventHistoryCardData,
  testimonialWithoutImageData,
  buyTicketCardData,
};
