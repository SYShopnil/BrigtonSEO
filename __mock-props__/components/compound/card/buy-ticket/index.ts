import { IBuyTicketCard } from '@src/types/compound';

export const buyTicketCardData: IBuyTicketCard = {
  title: 'Hurry Up!',
  description: 'Book Your Seat & learn From Industry Experts.',
  attendanceText: 'Total Attending',
  attendanceNumber: 3256,
  buttonInfo: {
    buttonText: 'buy ticket',
    link: 'https://www.google.com',
    linkOpenNewTab: false,
  },
};
