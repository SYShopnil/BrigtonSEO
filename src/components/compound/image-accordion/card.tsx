import { IImageAccordionCard } from '@src/types/compound';
import Image from 'next/image';
import Link from 'next/link';
import Styles from './index.module.scss';

interface IProps extends IImageAccordionCard {
  isActive: boolean;
  isLarge: boolean;
  onHover?: () => void;
}

/**
 *
 * @param {object}  IProps
 * @returns JSX.Element
 */
export const Card = ({
  image,
  title,
  link,
  description,
  isActive,
  isLarge,
  onHover,
}: IProps) => {
  /**
   * Tailwind classess
   */
  const titleText = `text-[1.5rem] font-semibold leading-[1.4] md:text-[1.75rem] text-white lg:text-[3rem] lg:font-bold lg:leading-[1.1]`;

  return (
    <div
      onMouseEnter={onHover}
      className={` ${isActive ? Styles['active'] : ''} ${
        isLarge ? Styles['large'] : Styles['small']
      }  ${Styles['card']}  relative overflow-hidden`}
    >
      {/* card overlay  */}
      <span
        className={`${Styles['card-overlay']} absolute inset-0 z-[1]`}
      ></span>
      <div className="inline-flex absolute w-full h-full">
        <Image src={image} alt={title} objectFit="cover" layout="fill" />
      </div>

      {/* only show for large card  */}
      {isLarge && (
        <h3 className={`${Styles['title-inactive']}  ${titleText}`}>
          <span>{title}</span>
        </h3>
      )}

      {/* some extra div taken for large card animation purpose */}
      <div
        className={`${Styles['content']} absolute left-0 bottom-0 p-4 lg:p-[3.75rem] z-10`}
      >
        <div className={`${Styles['content-inner']}`}>
          <div className={`${Styles['content-inner-inner']}`}>
            <h3 className={` ${Styles['title']} ${titleText}`}>
              <Link href={link}>
                <a>{title}</a>
              </Link>
            </h3>
            <p
              className={`${Styles['description']} text-sm lg:text-[1.125rem] lg:leading-[1.6] mt-4 text-white`}
            >
              {description}
            </p>
          </div>
        </div>
      </div>
    </div>
  );
};
