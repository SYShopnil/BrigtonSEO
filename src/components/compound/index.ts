import { ContactSection } from './contact-section';
import { ListContentSection } from './list-content-section';
import BuyTicketCard from './card/buy-ticket';
import { HeroGeneral, HeroHome } from './hero-section';
import CourseCard from './card/course';
import PricingCard from './card/pricing';
import { Testimonial } from './card/testimonial';
import Person from './card/person';
import UpcomingCard from './card/upcoming';
import FooterSection from './footer-section';
import { TestimonialSection } from './testimonial-section';
import { EventHistoryCard } from './card/event-history';
import { TextAccordion } from './textAccordion';
import { ImageAccordion } from './image-accordion';

export {
  ContactSection,
  CourseCard,
  FooterSection,
  ListContentSection,
  Testimonial,
  HeroGeneral,
  BuyTicketCard,
  PricingCard,
  Person,
  EventHistoryCard,
  UpcomingCard,
  TestimonialSection,
  HeroHome,
  TextAccordion,
  ImageAccordion,
};
