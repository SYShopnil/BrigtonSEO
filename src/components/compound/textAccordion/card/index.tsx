import { Card } from '@src/components/root';
import IconStore from '@src/components/root/_icons';
import {
  CardBorderRadiusEnum,
  CardShadowEnum,
  IconEnum,
} from '@src/types/root';
import React, { useState, useEffect, useRef } from 'react';
import Styles from '../index.module.scss';
import { ITextAccordion } from '@src/types/compound/textAccordion';

interface IAnswerPart extends Pick<ITextAccordion, 'answer'> {}
interface IQuestionPart
  extends Pick<ITextAccordion, 'question' | 'isSelected'> {}

const AnswerPart = ({ answer }: IAnswerPart) => {
  return <p className={`font-normal text-sm lg:text-base `}>{answer}</p>;
};

const QuestionPart = ({ question, isSelected }: IQuestionPart) => {
  const headingPartTextStyle: string = `font-normal lg:font-semibold text-base lg:text-xl`;
  const headingTextColor: string = isSelected
    ? Styles['text-active']
    : Styles['text-inactive'];
  return (
    <div className={`flex justify-between `}>
      <p className={`${headingPartTextStyle} ${headingTextColor} `}>
        {question}
      </p>
      {/* icon part  */}
      <div
        className={`flex justify-center items-center ${headingPartTextStyle} ${headingTextColor}`}
      >
        <IconStore
          iconName={
            isSelected
              ? IconEnum.AiOutlineMinusCircle
              : IconEnum.AiOutlinePlusCircle
          }
        />
      </div>
    </div>
  );
};

/**
 *
 * This component will control the opening and closing part of the accordion
 * It will show individual element of the following accordion
 * This is main component
 *
 */

export const AccordionCard = ({
  question,
  answer,
  isSelected,
  onCardSelect,
}: ITextAccordion) => {
  const cardShadow = isSelected
    ? CardShadowEnum.ShadowTwo
    : CardShadowEnum.None;
  const cardBorderRadius = isSelected
    ? CardBorderRadiusEnum['2XL']
    : CardBorderRadiusEnum.XL;

  const [height, setHeight] = useState<string>('0px');
  const answerWrapperRef: React.LegacyRef<HTMLDivElement> | undefined =
    useRef(null);

  /**
   *
   * This will control the selected accordion's toggle part
   *
   */
  useEffect(() => {
    if (isSelected) {
      setHeight(`${answerWrapperRef?.current?.scrollHeight}px`);
    } else {
      setHeight('0px');
    }
  }, [isSelected]);
  return (
    <Card
      dataTestId={'textAccordionListCard'}
      cardShadow={cardShadow}
      cardBorderRadius={cardBorderRadius}
    >
      <div
        className={`cursor-pointer  ${Styles[`activeBorder`]} `}
        onClick={onCardSelect}
      >
        {/* inner wrapper */}
        <div
          className={`my-[0.5rem] lg:my-[1.5rem] mx-[1rem] lg:mx-[1.5rem] space-y-[1rem]  lg:space-y-[1.5rem]`}
        >
          <QuestionPart question={question} isSelected={isSelected} />

          {/* here on select toggle animation part has applied by using ref */}
          <div
            className={`${
              Styles[`answer-active-animation`]
            } overflow-hidden transition-all duration-200 `}
            ref={answerWrapperRef}
            style={{ height: height }}
          >
            <AnswerPart answer={answer} />
          </div>
        </div>
      </div>
    </Card>
  );
};
