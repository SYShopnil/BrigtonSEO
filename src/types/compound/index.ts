import { IContactSection, IContact } from './contact-section';
import {
  IListItem,
  IListContentSection,
  IContentHead,
  ListStyleEnum,
  IListContentCtaButton,
} from './list-content-section';
import { ITestimonialSection } from './testimonial-section';
import { IAuthor } from './common';
import {
  IHeroGeneral,
  HeroBgColorEnum,
  IHeroHome,
  EventInfoThemeEnum,
  IHeroLogo,
  IInfoCount,
} from './hero-section';
import { ITestimonial } from './card/testimonial';
import {
  IPricingCard,
  PriceBoxColorSchemeEnum,
  PriceBoxSizeEnum,
} from './card/pricing';
import { IPerson, ITeam, IPersonTransparent } from './card/person';
import { IFooterSection } from './footer-section';
import { IBuyTicketCard } from './card/buy-ticket';
import { IEventHistoryCard } from './card/event-history';
import { ICourseCard, ITag, TagBackgroundEnum } from './card/course';
import type { IUpcomingCard } from './card/upcoming';
import { UpcomingCardThemeEnum } from './card/upcoming';
import { IEventDate, IEventInfo, UserIconSizeEnum } from './common';
import { SectionThemeEnum } from './common';
import { ITextAccordion, ITextAccordionList, IFAQ } from './textAccordion';
import { IImageAccordion, IImageAccordionCard } from './image-accordion';

export type {
  IContactSection,
  IContact,
  IListItem,
  IListContentSection,
  IContentHead,
  ITestimonial,
  IBuyTicketCard,
  ICourseCard,
  ITag,
  IHeroGeneral,
  IPerson,
  IUpcomingCard,
  IFooterSection,
  IListContentCtaButton,
  IHeroHome,
  IEventInfo,
  IHeroLogo,
  IInfoCount,
  IPricingCard,
  ITextAccordion,
  ITextAccordionList,
  IFAQ,
  ITeam,
  IPersonTransparent,
  IEventHistoryCard,
  ITestimonialSection,
  IAuthor,
  IEventDate,
  IImageAccordion,
  IImageAccordionCard,
};

export {
  HeroBgColorEnum,
  UpcomingCardThemeEnum,
  ListStyleEnum,
  EventInfoThemeEnum,
  PriceBoxColorSchemeEnum,
  PriceBoxSizeEnum,
  TagBackgroundEnum,
  UserIconSizeEnum,
  SectionThemeEnum,
};
