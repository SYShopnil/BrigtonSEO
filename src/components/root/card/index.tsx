/**
 * Internal Imports
 */
import { ICard } from '@src/types/root';
import Styles from './card.module.scss';

/**
 * Props
 */
interface Props extends ICard {
  children: JSX.Element;
}

/**
 *
 * @param {object}  Props
 * @returns JSX.Element
 * @description This component is a root card component which take a children component and render those here.
 */
export const Card = ({
  children,
  dataTestId,
  cardBorderRadius,
  cardShadow,
  hoverCardShadow,
}: Props) => {
  return (
    <div
      className={`relative overflow-hidden bg-white ${Styles['card']} ${
        Styles[cardBorderRadius]
      } ${Styles[cardShadow]} ${
        hoverCardShadow ? Styles[`hover-${hoverCardShadow}`] : ''
      }`}
      data-testid={`${dataTestId}`}
    >
      {children}
    </div>
  );
};
