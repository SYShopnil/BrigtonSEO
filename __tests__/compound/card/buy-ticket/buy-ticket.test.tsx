import { render } from '@testing-library/react';
import { BuyTicketCard } from '@src/components/compound';
import { buyTicketCardData } from '@root/__mock-props__/components/compound/card';

/**
 * Test if the buy ticket card is rendering correctly
 */
it('renders Buy Ticket Card correctly', () => {
  const { getByTestId } = render(<BuyTicketCard {...buyTicketCardData} />);
  const getCardRootElement = getByTestId('buy-ticket-card');
  expect(getCardRootElement).toBeInTheDocument();
  expect(getCardRootElement).toMatchSnapshot();
});
