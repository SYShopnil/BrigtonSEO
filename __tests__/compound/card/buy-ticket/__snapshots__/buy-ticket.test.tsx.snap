// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Buy Ticket Card correctly 1`] = `
<div
  class="relative overflow-hidden bg-white card radius-2xl shadow-Two "
  data-testid="buy-ticket-card"
>
  <div
    class="flex md:flex-row flex-col md:px-10 md:py-6 px-3 py-4 max-w-[38rem]"
  >
    <div
      class="flex flex-[0_0_55%] flex-col md:pb-0 md:pr-[2.5rem] pb-[1.5rem] pr-0"
    >
      <div
        class="flex flex-col"
      >
        <h3
          class=" md:text-3xl text-xl text-primary pb-2 font-semibold md:self-start self-center md:text-left text-center"
        >
          Hurry Up!
        </h3>
        <p
          class="md:text-base text-sm text-black md:item-start item-center md:text-left text-center"
        >
          Book Your Seat & learn From Industry Experts.
        </p>
      </div>
      <div
        class="md:pt-[1.875rem] pt-4 md:flex-none pl-0 flex-[0_0_auto] md:self-start self-center"
      >
        <a
          class="inline-flex"
          href="https://www.google.com"
          target="_self"
        >
          <span
            class="bs-button BlueBgTextWhite hover-BlueVerticalGradientBgTextWhite btn-sm   group relative cursor-pointer overflow-hidden inline-flex gap-[1rem] items-center justify-center capitalize"
            data-testid="redirect-button"
          >
            <span>
              buy ticket
            </span>
            <span
              class="absolute bs-button-bg inset-0"
            />
          </span>
        </a>
      </div>
    </div>
    <span
      class="flex-[0_0_auto] md:w-[1px] w-1/2 max-sm:h-[1px] max-sm:mx-auto bg-[rgba(12,94,232,0.16)]"
    />
    <div
      class="flex-[0_0_40%] md:text-left md:pl-[2.5rem] text-center self-center pl-0"
    >
      <h6
        class="md:text-lg text-sm tracking-[0.5px] md:pt-0 pt-[1.5rem] pb-1 text-blackLight"
      >
        Total Attending
      </h6>
      <div
        class="md:text-5xl text-[3.5rem] leading-[4.2rem] font-bold text-secondary inline-flex items-center"
      >
        3,256
      </div>
    </div>
  </div>
</div>
`;
